### PR TITLE
make the compiler a bit easier on itself

### PR DESCRIPTION
**Pull request by JeffBezanson**
mostly by using higher-order functions less. This gives some further package load and compile time improvements.

### CODE REVIEW

File: base/compiler/optimize.jl [hunk 1]
```diff
@@ -143,7 +143,7 @@ function optimize(opt::OptimizationState, @nospecialize(result))
     def = opt.linfo.def
     nargs = Int(opt.nargs) - 1
     @timeit "optimizer" ir = run_passes(opt.src, nargs, opt)
-    force_noinline = any(x -> isexpr(x, :meta) && x.args[1] == :noinline, ir.meta)
+    force_noinline = _any(x -> isexpr(x, :meta) && x.args[1] == :noinline, ir.meta)
```
Comment (vtjnash) [inline base/compiler/optimize.jl]: Also nospecialize?

File: base/compiler/ssair/domtree.jl [hunk 2]
```diff
@@ -82,7 +82,7 @@ function construct_domtree(cfg::CFG)
         for dom in doms
             i == dom && continue
             let i = i, dom = dom
-                any(p -> (p !== i && p !== dom && dom in dominators[p]), doms) && continue
+                _any(p -> (p !== i && p !== dom && dom in dominators[p]), doms) && continue
```
Comment (vtjnash) [inline base/compiler/ssair/domtree.jl]: I’m not sure this is needed - these are all integers and IntSet operations

**Review by vtjnash — APPROVED** [review]

**Comment by tkoolen** [comment]
(haven't profiled, but) could `in` be another candidate for this kind of change?

https://github.com/JuliaLang/julia/blob/656d58782c5ae0fad3b66f430c2e1e50915ae468/base/reduce.jl#L710

**Comment by JeffBezanson** [comment]
Yes, quite possibly.

**Pull request merged**